workflow/controller/dag.go

```diff
@@ -209,7 +209,11 @@ func (d *dagContext) assessDAGPhase(targetTasks []string, nodes wfv1.Nodes, isSh
 			// their "depends" clause during their respective "dependencies" to "depends" conversion. See "expandDependency"
 			// in ancestry.go
 			if task := d.GetTask(depName); task.ContinuesOn(branchPhase) {
```

Comment:
I wonder if `continueOn` is even relevant at this point? If its parent failed, it doesn't matter if it has `continueOn`

Reply:
Yes, I think we can delete this piece of `contineOn` logic, now I added `TODO`